### PR TITLE
Document random options

### PR DESCRIPTION
**Pull request by appgurueu**
This was collecting dust on my solid state drive, so I decided to give it some finishing touches and put it up for review.

### CODE REVIEW

**Review by wsor4035 — APPROVED** [review]
good enough

File: doc/random.adoc [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+===== `:next([min, max])`
+
+If `min` and `max` are both omitted,
+they default to `-2^31` (`-2147483648`)
+and `2^31 - 1` (`2147483647`) respectively.
+
+===== `:rand_normal_dist(min, max, [num_trials])`
+
+WARNING: No successful use of this function is documented. Consider implementing your own normal distribution instead.
```
Comment (rollerozxa) [inline doc/random.adoc:84]: Usage in mods: https://web.archive.org/web/20231224131713if_/https://content.minetest.net/zipgrep/1b9f9e78-99da-4c02-9f75-f6134f362fa7/

File: doc/random.adoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+WARNING: Requires `((max - min) == 32767) or ((max-min) <= 6553))` for a proper distribution.
+
+=== `SecureRandom`
+
+System-provided cryptographically secure random:
+An attacker should not be able to predict the generated sequence of random numbers.
+Use this when generating cryptographic keys or tokens.
+
+Not necessarily available in all builds and on all platforms.
```
Comment (rollerozxa) [inline doc/random.adoc:119]: Probably should elaborate on this. On Windows it uses Win32's crypto API, and on every other platform it uses `/dev/urandom` which should exist on everywhere else (including Android).

File: doc/random.adoc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+== Comparison
+
+|===
+| Random Source  | Performance      | Bytes of entropy | Seedability                    | Versatility | Distribution | Security | Portability
+| `math.random`  | very good (1x)   | up to 4          | global seed; seeded by default | very good   | no guarantees, but usually decent enough | not cryptographically secure | varies by platform
+| `PcgRandom`    | okay (~14x)      | up to 4          | per-instance seed | very good   | good, decent guarantees | not cryptographically secure | always the same
+| `PseudoRandom` | okay (~15x)      | 1 to 2           | per-instance seed | outright sucks | okay-ish | not cryptographically secure | always the same
+| `SecureRandom` | still okay (30x) | 1 to 2048        | not seedable | cryptographically secure | varies by platform; may be missing
```
Comment (rollerozxa) [inline doc/random.adoc:175]: too few columns on this row? markdown-style tables are messy with so many columns when they are not properly formatted, [asciidoc supports a more ergonomic table syntax](https://github.com/minetest/minetest_docs/pull/17/files#diff-ee860ca877adc533b0369fb3fa42a24943010a41f40d49c84aa760da1ee0f49fR11) that's a good idea to use in these cases

**Pull request merged**